src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -3735,22 +3735,24 @@ static void Main()
         public void CreateTupleTypeSymbol_BadArguments()
         {
             var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef }); // no ValueTuple
-            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            TypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
 
-            Assert.Throws<ArgumentNullException>(() => comp.CreateTupleTypeSymbol(default(ImmutableArray<ITypeSymbol>), default(ImmutableArray<string>)));
-
-            // 0-tuple and 1-tuple are not supported at this point
-            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol>.Empty, default(ImmutableArray<string>)));
-            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType }.AsImmutable(), default(ImmutableArray<string>)));
+            Assert.Throws<ArgumentNullException>(() => comp.CreateTupleTypeSymbol(null, default(ImmutableArray<string>)));
 
             // if names are provided, you need one for each element
-            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType, intType }.AsImmutable(), new[] { "Item1" }.AsImmutable()));
+            var vt2 = comp.GetWellKnownType(WellKnownType.System_ValueTuple_T2).Construct(intType, intType);
+            try
+            {
+                comp.CreateTupleTypeSymbol(vt2, new[] { "Item1" }.AsImmutable());
+                Assert.True(false);
+            }
+            catch (ArgumentException e)
+            {
+                Assert.Contains(CodeAnalysisResources.TupleNamesAllOrNone, e.Message);
+            }
```

Comment:
I wanted to check and differentiate the different cases that throw `ArgumentException`: if the type is not C# type (see another test below), or if it is not tuple compatible.